framework/src/userobjects/ElementSubdomainModifier.C

```diff
@@ -445,3 +603,26 @@ ElementSubdomainModifier::setOldAndOlderSolutionsForMovedNodes(SystemBase & sys)
   if (older_solution)
     older_solution->close();
 }
+
+void
+ElementSubdomainModifier::setAncestorsSubdomainIDs(const SubdomainID & subdomain_id,
+                                                   const dof_id_type & elem_id)
+{
+  Elem * curr_elem = _mesh.elemPtr(elem_id);
+
+  unsigned int lv = curr_elem->level();
+
+  for (unsigned int i = lv; i > 0; --i)
+  {
+    // Change the parent's subdomain, if any
+    curr_elem = curr_elem->parent();
+    dof_id_type id = curr_elem->id();
+    Elem * elem = _mesh.elemPtr(id);
+    elem->subdomain_id() = subdomain_id;
```

Comment:
What is the ancestor's subdomain based on if the refined element is split?

Reply:
It will be the subdomain that we are moving the elements to. This is kinda specialized to the AM simulations where we do not want to accidentally remove any volume during coarsening.